allensdk/internal/api/behavior_data_lims_api.py

```diff
@@ -0,0 +1,465 @@
+            how="left")
+        if len(raw_stim_pres_df) != len(stim_pres_df):
+            raise ValueError("Length of `stim_pres_df` should not change after"
+                             f" merge; was {len(raw_stim_pres_df)}, now "
+                             f" {len(stim_pres_df)}.")
+        return stim_pres_df[sorted(stim_pres_df)]
+
+    @memoize
+    def get_stimulus_templates(self) -> Dict[str, np.ndarray]:
+        """Get stimulus templates (movies, scenes) for behavior session.
+
+        Returns
+        -------
+        Dict[str, np.ndarray]
+            A dictionary containing the stimulus images presented during the
+            session. Keys are data set names, and values are 3D numpy arrays.
+        """
+        data = self._behavior_stimulus_file()
+        return get_stimulus_templates(data)
+
+    @memoize
+    def get_stimulus_timestamps(self) -> np.ndarray:
+        """Get stimulus timestamps (vsyncs) from pkl file.
+
+        NOTE: Located with behavior_session_id. Does not use the sync_file
+        which requires ophys_session_id.
+
+        Returns
+        -------
+        np.ndarray
+            Timestamps associated with stimulus presentations on the monitor
+            that do no account for monitor delay.
+        """
+        data = self._behavior_stimulus_file()
+        vsyncs = data["items"]["behavior"]["intervalsms"]
+        return np.hstack((0, vsyncs)).cumsum() / 1000.0  # cumulative time
+
+    def get_task_parameters(self) -> dict:
+        """Get task parameters from pkl file.
+
+        Returns
+        -------
+        dict
+            A dictionary containing parameters used to define the task runtime
+            behavior.
+        """
+        data = self._behavior_stimulus_file()
+        return get_task_parameters(data)
+
+    def get_trials(self) -> pd.DataFrame:
+        """Get trials from pkl file
+
+        Returns
+        -------
+        pd.DataFrame
+            A dataframe containing behavioral trial start/stop times,
+            and trial data
+        """
+        licks = self.get_licks()
+        data = self._behavior_stimulus_file()
+        rewards = self.get_rewards()
+        stimulus_presentations = self.get_stimulus_presentations()
+        # Pass a dummy rebase function since we don't have two time streams
+        trial_df = get_trials(data, licks, rewards, stimulus_presentations,
+                              lambda x: x)
+
+        return trial_df
+
+    @memoize
+    def get_sex(self) -> str:
+        """Returns sex of the animal (M/F)
+        :rtype: str
+        """
+        query = f"""
+            SELECT g.name AS sex
+            FROM behavior_sessions bs
+            JOIN donors d ON bs.donor_id = d.id
+            JOIN genders g ON g.id = d.gender_id
+            WHERE bs.id = {self.behavior_session_id};
+            """
+        return self.fetchone(query, strict=True)
+
+    @memoize
+    def get_age(self) -> str:
```

Comment:
This will get the age at death of the mouse.  If the mouse is still alive, it just returns 'unknown'.  I think it is probably more useful to get the age in days at the time of the behavior session.  You could get that by subtracting the birth date from the time of the experiment.
```
SELECT get_experiment_date(self) MINUS donors.date_of_birth AS age_in_days
FROM behavior_sessions bs
JOIN donors d ON d.id = bs.donor_id
WHERE bs.id = {self.behavior_session_id};
```

Reply:
I'm mirroring the age query that is currently implemented in BehaviorOphysLimsApi (below) to preserve expected behavior:

                ```SELECT a.name as age
                FROM ophys_experiments oe
                JOIN ophys_sessions os ON oe.ophys_session_id = os.id
                JOIN specimens sp ON sp.id=os.specimen_id
                JOIN donors d ON d.id=sp.donor_id
                JOIN ages a ON a.id=d.age_id
                WHERE oe.id={self.ophys_experiment_id}```

Is the age in days query you described the behavior that scientists want instead?

Reply:
From conversation with Nick P, let's leave the get_age method as you originally wrote it, but add another method to add ability to get birth date.  Then user, can easily subtract birth date from experiment date to get age of mouse at time of experiment.

Reply:
Updated, added `get_birth_date` method.